base/utils/R/utils.R

```diff
@@ -209,7 +209,7 @@ summarize.result <- function(result) {
     dplyr::filter(n == 1) %>%
     dplyr::group_by(citation_id, site_id, trt_id,
                     control, greenhouse, date, time,
-                    cultivar_id, specie_id) %>%
+                    cultivar_id, specie_id, name, treatment_id) %>%
```

Comment:
I think this is where it complains. @infotroph how should this be fixed?

Reply:
no idea but perhaps it is related to this https://www.r-bloggers.com/2019/08/no-visible-binding-for-global-variable/ and also covered here: https://community.rstudio.com/t/how-to-solve-no-visible-binding-for-global-variable-note/28887

![image](https://user-images.githubusercontent.com/464871/102409622-9e700b80-3fac-11eb-8a26-71c0bddaa7c4.png)

which implies that it could be fixed by changing this to 

```r
dplyr::group_by(.data$citation_id, .data$site_id, .data$trt_id,
                    .data$control, .data$greenhouse, .data$date, .data$time,	.data$control, .data$greenhouse, .data$date, .data$time,
                    .data$cultivar_id, .data$specie_id) %>%	                    cultivar_id, specie_id, name, treatment_id)
```

but its not clear why only the two new variables are called out in the NOTE. Or if we have to start prepending variables everywhere with `.data$` 

and FWIW this is a note - note even a warning

Reply:
Well I just corrected them to include .data$, but it has still failed that check.

Reply:
@dlebauer beat me to my answer: `.data` is indeed what you need here, and the reason it's needed is that `R CMD check`'s static scope analysis doesn't know about nonstandard evaluation, so it can't tell that these variables will be provided by the dataframe.

As for why the NOTE only calls out the two newly added ones: `R CMD check` actually does complain about all of them, but they get special exceptions because they already existed before I wrote the CI script. For gory details, see section 8.4.3, "Continuous Integration", in the [developer guide](https://pecanproject.github.io/pecan-documentation/develop/developer-testing.html). 

Now that you've fixed this NOTE, the special exceptions are no longer needed and you can remove them by deleting all the lines starting `summarize.result: no visible binding for global variable` in `base/utils/test/Rcheck_reference.log`